### PR TITLE
feat(transformer): implement @babel/plugin-transform-react-jsx-source

### PR DESCRIPTION
**Pull request by fireairforce**
imeplement: https://github.com/babel/babel/tree/main/packages/babel-plugin-transform-react-jsx-source

### CODE REVIEW

File: crates/oxc_transformer/src/react_jsx/jsx_src.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      ))
+    );
+    // lineNumber
+    properties.push(
+      ObjectPropertyKind::ObjectProperty(self.ctx.ast.object_property(
+        SPAN, 
+        kind, 
+        self.ctx.ast.property_key_identifier(IdentifierName::new(SPAN, "lineNumber".into())),
+        // TODO: i am not sure it there any way for me to know the lineNumber.. maybe by SPAN 
+        self.ctx.ast.literal_number_expression(self.ctx.ast.number_literal(SPAN, 10.0, "10", NumberBase::Decimal)),
```
Comment (fireairforce) [inline crates/oxc_transformer/src/react_jsx/jsx_src.rs:49]: I am not sure how to get the lineNumber here, maybe i can calcute by `SPAN`?

**Comment by Dunqing** [comment]
Thank you for your contribution! We plan to re-implement the transformer. So we will not consider merging any transformer PRs for now. Learn more at https://github.com/oxc-project/oxc/discussions/2704

**Comment by fireairforce** [comment]
ok, thx for your reply, looking forward for the re-implement~ maybe i can contribute after the re-implement done.

**Comment by Boshen** [comment]
Thank you for working on this. I copied your implementation and merged them in https://github.com/oxc-project/oxc/pull/2948

**Pull request closed without merging**